### PR TITLE
feat: move mypy config to pyproject.toml

### PR DESCRIPTION
**Pull request by frostming**
Signed-off-by: Frost Ming <me@frostming.com>

So that users can run mypy locally with consistent behavior

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -10,6 +10,7 @@ dependencies = [
     "flask>=3.0.0",
     "libcst>=1.1.0",
     "gunicorn>=21.2.0",
+    "typing-extensions>=4.8.0",
```
Comment (laike9m) [inline pyproject.toml:13]: Why do we need this (since we require Python >= 3.12 which is the latest version)

**Review by laike9m — APPROVED** [review]
LGTM except one question about the extra dependency

File: pyproject.toml (hunk 1 above)
Comment (frostming) [reply pyproject.toml:13]: `typing-extensions` should always include the latest changes in Python Typing. No harm to explicitly specify it here(in fact it is already installed by other dependents).

BTW, does zeabur non-Dockerfile deployment respect the `requires-python` value?

Comment (laike9m) [reply pyproject.toml:13]: okay I can buy that argument.

> BTW, does zeabur non-Dockerfile deployment respect the requires-python value?
Not sure, I'm asking them

**Pull request merged**